src/main/java/com/atlauncher/viewmodel/impl/settings/GeneralSettingsViewModel.java

```diff
@@ -0,0 +1,431 @@
+            if (getDateFormats()[index].equals(App.settings.dateFormat))
+                _addOnDateFormatChanged.onNext(index);
+        }
+    }
+
+    @Override
+    public String[] getInstanceTitleFormats() {
+        return Constants.INSTANCE_TITLE_FORMATS;
+    }
+
+    @Override
+    public void setInstanceTitleFormat(String format) {
+        App.settings.instanceTitleFormat = format;
+        SettingsManager.post();
+    }
+
+    @Override
+    public Observable<Integer> InstanceFormatChanged() {
+        return _addOnInstanceFormatChanged.observeOn(SwingSchedulers.edt());
+    }
+
+    private void pushInstanceFormat() {
+        for (int index = 0; index < getInstanceTitleFormats().length; index++) {
+            if (getInstanceTitleFormats()[index].equals(App.settings.instanceTitleFormat))
+                _addOnInstanceFormatChanged.onNext(index);
+        }
+    }
+
+    @Override
+    public void setSelectedTabOnStartup(int tab) {
+        App.settings.selectedTabOnStartup = tab;
+        SettingsManager.post();
+    }
+
+    @Override
+    public Observable<Integer> SelectedTabOnStartupChanged() {
+        return _addOnSelectedTabOnStartupChanged.observeOn(SwingSchedulers.edt());
+    }
+
+    @Override
+    public InstanceSortingStrategies[] getInstanceSorting() {
+        return InstanceSortingStrategies.values();
+    }
+
+    @Override
+    public void setInstanceSorting(InstanceSortingStrategies sorting) {
+        App.settings.defaultInstanceSorting = sorting;
+        SettingsManager.post();
+    }
+
+    @Override
+    public Observable<Integer> addInstanceSortingChanged() {
+        return _addInstanceSortingChanged.observeOn(SwingSchedulers.edt());
+    }
+
+    private void pushInstanceSorting() {
+        for (int index = 0; index < getInstanceSorting().length; index++) {
+            if (getInstanceSorting()[index] == App.settings.defaultInstanceSorting)
+                _addInstanceSortingChanged.onNext(index);
+        }
+    }
+
+    @Override
+    public void resetCustomDownloadPath() {
+        App.settings.customDownloadsPath = null;
+        SettingsManager.post();
+    }
+
+    @Override
+    public void setCustomsDownloadPath(String value) {
+        App.settings.customDownloadsPath = value;
+        SettingsValidityManager.setValidity("customDownloadsPath", true);
+        SettingsManager.post();
+    }
+
+    @Override
+    public void setCustomsDownloadPathPending() {
+        SettingsValidityManager.setValidity("customDownloadsPath", false);
+    }
+
+    @Override
+    public Observable<String> CustomsDownloadPathChanged() {
+        return _addOnCustomsDownloadPathChanged.observeOn(SwingSchedulers.edt());
+    }
+
+    @Override
+    public void setKeepLauncherOpen(boolean b) {
+        App.settings.keepLauncherOpen = b;
+        SettingsManager.post();
+    }
+
+    @Override
+    public Observable<Boolean> KeepLauncherOpenChanged() {
```

Comment:
A lot of these are PascalCase rather than camelCase. Guessing this is unintentional from copy paste?

Reply:
Yep, lots of edits, can fix